packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -104,13 +183,15 @@ class CupertinoTabScaffold extends StatefulWidget {
   /// that lets the user switch between different tabs in the main content area
   /// when present.
   ///
-  /// Setting and changing [CupertinoTabBar.currentIndex] programmatically will
-  /// change the currently selected tab item in the [tabBar] as well as change
-  /// the currently focused tab from the [tabBuilder].
-
+  /// The [CupertinoTabBar.currentIndex] property of [tabBar] is *overridden*
```

Comment:
The [CupertinoTabBar.currentIndex] property of the [tabBar] is ignored when a [controller] is provided.